test-suite-groovy/src/test/groovy/io/micronaut/kafka/docs/seek/aware/ProductListener.groovy

```diff
@@ -0,0 +1,37 @@
+package io.micronaut.kafka.docs.seek.aware
+
+import io.micronaut.configuration.kafka.ConsumerSeekAware
+import io.micronaut.configuration.kafka.annotation.*
+import io.micronaut.configuration.kafka.seek.*
+import io.micronaut.context.annotation.Requires
+import io.micronaut.kafka.docs.Product
+import jakarta.inject.Inject
+import org.apache.kafka.common.TopicPartition
+
+@KafkaListener
+@Requires(property = "spec.name", value = "ConsumerSeekAwareSpec")
+class ProductListener implements ConsumerSeekAware { // <1>
+
+    List<Product> processed = []
+
+    @Inject
```

Comment:
why the `@Inject`? The `@Inject` should be necessary only we we had multiple constructors and we wanted to signal the constructor to be used by the framework to instantiate the class.  Why the empty constructor?

Reply:
@sdelamo 

I know `@Inject` is not strictly necessary. It was there to let the user know that the config object will be _injected_ by Micronaut. But don't worry, I'll remove it.

The empty constructor is there because I need `ProductListenerConfiguration` to produce the test products _before_ `ProductListener` rebalances. This dependency ensures that things happen in a deterministic order.